### PR TITLE
Makes Combat Mechs moderately harder to create.

### PR DESCRIPTION
**Pull request by Dsdiy**
## What Does This PR Do
Removes robotics ability to mass produce combat mechs easily by locking the boards behind cargo. (Utility mechs such as ripley, odysseus, etc. are unchanged. Phazon is unchanged because it's already hard enough to get)

## Why It's Good For The Game
Robotics being able to mass produce combat mechs with little to no difficulty causes them to absolutely stomp on midrounds such as blob, terrors, etc. This will at least make them have to go to cargo to get the boards, wait on them to be delivered, and head all the way back to robotics to finish their mech, buying midrounds valuable time to get going.

## Testing
Ran it on my custom server, everything worked as intended. The boards delivered properly, and weren't printable from RnD.

## Changelog
:cl:
Removed RnD's ability to print combat mech boards.
Added combat mech board crates to Cargo.
/:cl:

### CODE REVIEW

File: code/datums/mecha_designs.dm [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (APLU \"Ripley\" Central Control module)"
+	desc = "Allows for the construction of a \"Ripley\" Central Control module."
+	id = "ripley_main"
+	req_tech = list("programming" = 2)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/ripley/main
+	category = list("Exosuit Modules")
+
+/datum/design/ripley_peri
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:15]:
```suggestion
/datum/design/ripley_peripherals
```
Please avoid shortening words like this, you may now know what `peri` stands for, but future coders won't. See how the build path is similar: `/obj/item/circuitboard/mecha/ripley/peripherals`

File: code/datums/mecha_designs.dm [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"Odysseus\" Central Control module)"
+	desc = "Allows for the construction of a \"Odysseus\" Central Control module."
+	id = "odysseus_main"
+	req_tech = list("programming" = 3,"biotech" = 3, "engineering" = 3)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/odysseus/main
+	category = list("Exosuit Modules")
+
+/datum/design/odysseus_peri
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:36]:
```suggestion
/datum/design/odysseus_peripherals
```

File: code/datums/mecha_designs.dm [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"Gygax\" Central Control module)"
+	desc = "Allows for the construction of a \"Gygax\" Central Control module."
+	id = "gygax_main"
+	req_tech = list("programming" = 8, "combat" = 8, "engineering" = 8)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/gygax/main
+	category = list("Exosuit Modules")
+
+/datum/design/gygax_peri
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:57]:
```suggestion
/datum/design/gygax_peripherals
```

File: code/datums/mecha_designs.dm [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"Gygax\" Peripherals Control module)"
+	desc = "Allows for the construction of a \"Gygax\" Peripheral Control module."
+	id = "gygax_peri"
+	req_tech = list("programming" = 8, "combat" = 8, "engineering" = 8)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/gygax/peripherals
+	category = list("Exosuit Modules")
+
+/datum/design/gygax_targ
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:67]:
```suggestion
/datum/design/gygax_targeting
```

File: code/datums/mecha_designs.dm [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"Durand\" Central Control module)"
+	desc = "Allows for the construction of a \"Durand\" Central Control module."
+	id = "durand_main"
+	req_tech = list("programming" = 8, "combat" = 8, "engineering" = 8)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/durand/main
+	category = list("Exosuit Modules")
+
+/datum/design/durand_peri
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:88]:
```suggestion
/datum/design/durand_peripherals
```

File: code/datums/mecha_designs.dm [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"Durand\" Peripherals Control module)"
+	desc = "Allows for the construction of a \"Durand\" Peripheral Control module."
+	id = "durand_peri"
+	req_tech = list("programming" = 8, "combat" = 8, "engineering" = 8)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/durand/peripherals
+	category = list("Exosuit Modules")
+
+/datum/design/durand_targ
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:98]:
```suggestion
/datum/design/durand_targeting
```

File: code/datums/mecha_designs.dm [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"Phazon\" Central Control module)"
+	desc = "Allows for the construction of a \"Phazon\" Central Control module."
+	id = "phazon_main"
+	req_tech = list("programming" = 6, "materials" = 6, "plasmatech" = 5)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000, MAT_BLUESPACE = 100)
+	build_path = /obj/item/circuitboard/mecha/phazon/main
+	category = list("Exosuit Modules")
+
+/datum/design/phazon_peri
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:119]:
```suggestion
/datum/design/phazon_peripherals
```

File: code/datums/mecha_designs.dm [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"Phazon\" Peripherals Control module)"
+	desc = "Allows for the construction of a \"Phazon\" Peripheral Control module."
+	id = "phazon_peri"
+	req_tech = list("programming" = 6, "bluespace" = 5, "plasmatech" = 5)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000, MAT_BLUESPACE = 100)
+	build_path = /obj/item/circuitboard/mecha/phazon/peripherals
+	category = list("Exosuit Modules")
+
+/datum/design/phazon_targ
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:129]:
```suggestion
/datum/design/phazon_targeting
```

File: code/datums/mecha_designs.dm [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"H.O.N.K\" Central Control module)"
+	desc = "Allows for the construction of a \"H.O.N.K\" Central Control module."
+	id = "honker_main"
+	req_tech = list("programming" = 3, "engineering" = 3)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/honker/main
+	category = list("Exosuit Modules")
+
+/datum/design/honker_peri
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:150]:
```suggestion
/datum/design/honker_peripherals
```

File: code/datums/mecha_designs.dm [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Board (\"H.O.N.K\" Peripherals Control module)"
+	desc = "Allows for the construction of a \"H.O.N.K\" Peripheral Control module."
+	id = "honker_peri"
+	req_tech = list("programming" = 3, "engineering" = 3)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/honker/peripherals
+	category = list("Exosuit Modules")
+
+/datum/design/honker_targ
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:160]:
```suggestion
/datum/design/honker_targeting
```

File: code/datums/mecha_designs.dm [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Module (\"Reticence\" Central Control module)"
+	desc = "Allows for the construction of a \"Reticence\" Central Control module."
+	id = "reticence_main"
+	req_tech = list("programming" = 3, "engineering" = 3)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/reticence/main
+	category = list("Exosuit Modules")
+
+/datum/design/reticence_peri
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:180]:
```suggestion
/datum/design/reticence_peripherals
```

File: code/datums/mecha_designs.dm [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+	name = "Exosuit Module (\"Reticence\" Peripherals Control module)"
+	desc = "Allows for the construction of a \"Reticence\" Peripheral Control module."
+	id = "reticence_peri"
+	req_tech = list("programming" = 3, "engineering" = 3)
+	build_type = IMPRINTER
+	materials = list(MAT_GLASS = 1000)
+	build_path = /obj/item/circuitboard/mecha/reticence/peripherals
+	category = list("Exosuit Modules")
+
+/datum/design/reticence_targ
```
Comment (Miraviel) [inline code/datums/mecha_designs.dm:190]:
```suggestion
/datum/design/reticence_targeting
```

File: code/modules/research/designs/mecha_designs.dm [hunk 13]
```diff
@@ -48,7 +48,7 @@
 	name = "Exosuit Board (\"Gygax\" Central Control module)"
 	desc = "Allows for the construction of a \"Gygax\" Central Control module."
 	id = "gygax_main"
-	req_tech = list("programming" = 4, "combat" = 3, "engineering" = 4)
+	req_tech = list("programming" = 8, "combat" = 8, "engineering" = 8)
```
Comment (Miraviel) [inline code/modules/research/designs/mecha_designs.dm:51]: Programming 8 is currently achievable by:
- `/obj/item/abductor/silencer` (agent abductor)
- `/obj/item/deactivated_swarmer` (swarmer spawner)
- `/obj/item/implant/weapons_auth` (currently unobtainable outside the syndicate research base)

Combat 8:
- `/obj/item/gun/energy/laser/instakill` (unobtainable)
- `/obj/item/gun/projectile/automatic/sniper_rifle` (only(?) comes from wizard summon guns spell)
- `/obj/item/gun/projectile/automatic/sniper_rifle/syndicate` (nuclear operative item)
- `/obj/item/organ/internal/cyberimp/arm/combat` (unobtainable)

Engineering 8 comes from:
- `/obj/item/abductor/gizmo` (scientist abductor)
- `/obj/item/assembly/signaler/anomaly/vortex` (anomaly event)

With this change, circuits are pretty much impossible to be ever achieved.

**Review by Miraviel — CHANGES_REQUESTED** [review]

File: code/modules/research/designs/mecha_designs.dm (hunk 13 above)
Comment (Dsdiy) [reply code/modules/research/designs/mecha_designs.dm:51]: This is more to make them unable to be created via RnD, but still easily obtainable via cargo.

**Comment by Qwertytoforty** [comment]
As suggested the last time(s) this came up. I would suggest making mechs instead require a power core (or some other item) 
orderable through cargo, while having robotics and/ or RD start with one. This allows crew to make one (or 2) mechs, and then pay a steep cargo price for more. This means crew can still build one in a time of crisis, and an antag can make one more stealthily without having to order the board from cargo

File: code/modules/research/designs/supplypacks.dm [hunk 14]
```diff
@@ -0,0 +1,2033 @@
+//SUPPLY PACKS
```
Comment (Cocacolagua) [inline code/modules/research/designs/supplypacks.dm:1]: you seem to have accidentally duped this file

Comment (Dsdiy) [reply code/modules/research/designs/supplypacks.dm:1]: I'm honestly not even sure how I managed to pull this off or how to fix this.

**Comment by AffectedArc07** [comment]
> Your PR has been put on hold as your GitHub account is not linked to a Paradise forums account.
> 
> Please link your account at https://www.paradisestation.org/forum/custom/link_github_account.php
> 
> _Headcoders may ignore this in specific circumstances (Minor fix PR, trusted contributor from another codebase, anything at their discretion)_

PR author has now linked account

File: code/datums/mecha_designs.dm [hunk 15]
```diff
@@ -0,0 +1,198 @@
+///////////////////////////////////
```
[the remark this answers is not in the document]
Comment (Dsdiy) [reply code/datums/mecha_designs.dm:1]: I'm still not experienced in coding, so I'm not 100% sure which part of the file your talking about.  Could you explain what part please?

[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/datums/mecha_designs.dm:1]: So in `code/modules/research/designs/mecha_designs.dm` you have

```dm
/datum/design/ripley_peripherals
	name = "Exosuit Board (APLU \"Ripley\" Peripherals Control module)"
	desc = "Allows for the construction of a  \"Ripley\" Peripheral Control module."
	id = "ripley_peri"
	req_tech = list("programming" = 2)
	build_type = IMPRINTER
	materials = list(MAT_GLASS = 1000)
	build_path = /obj/item/circuitboard/mecha/ripley/peripherals
	category = list("Exosuit Modules")
```

And in this file (`code/datums/mecha_designs.dm`) you have:
```dm
/datum/design/ripley_peri
	name = "Exosuit Board (APLU \"Ripley\" Peripherals Control module)"
	desc = "Allows for the construction of a  \"Ripley\" Peripheral Control module."
	id = "ripley_peri"
	req_tech = list("programming" = 2)
	build_type = IMPRINTER
	materials = list(MAT_GLASS = 1000)
	build_path = /obj/item/circuitboard/mecha/ripley/peripherals
	category = list("Exosuit Modules")
```

You have duplicated the code for all the designs, given them the same ID, when the existing ones should just be tweaked.

[the remark this answers is not in the document]
Comment (Dsdiy) [reply code/datums/mecha_designs.dm:1]: Alright, I'll tweak their ID's. Thought I was forgetting something.

[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/datums/mecha_designs.dm:1]: No thats not the point. My point is why have you made the exact same code twice with 0 change

**Comment by Dsdiy** [comment]
After some debates with other contributors, I feel this isn't the way to go for nerfing combat mech production. I'll close this for now and start working on Qwerty's suggestion.

**Pull request closed without merging**